### PR TITLE
chore(ctb): Update `dispute` contracts' comment style

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Updates the doc comment style of all `dispute` related contracts to adhere to the style guide.

### CODE REVIEW

**Comment by clabby** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #6103** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/monorepo/6103" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/monorepo/6103?utm_source=stack-comment).

File: packages/contracts-bedrock/contracts/dispute/interfaces/IDisputeGameFactory.sol [hunk 1]
```diff
@@ -5,105 +5,87 @@ import "../../libraries/DisputeTypes.sol";
...
+    /// @notice Emitted when a new game implementation added to the factory
+    /// @param impl The implementation contract for the given `GameType`.
+    /// @param gameType The type of the DisputeGame.
     event ImplementationSet(address indexed impl, GameType indexed gameType);
 
-    /**
-     * @notice the total number of dispute games created by this factory.
-     * @return _gameCount The total number of dispute games created by this factory.
-     */
+    /// @notice the total number of dispute games created by this factory.
```
Comment (refcell) [inline packages/contracts-bedrock/contracts/dispute/interfaces/IDisputeGameFactory.sol:26]: Small nit, keep start of `@notice` tags capitalized
```suggestion
    /// @notice The total number of dispute games created by this factory.
```

File: packages/contracts-bedrock/contracts/dispute/interfaces/IDisputeGameFactory.sol [hunk 2]
```diff
@@ -5,105 +5,87 @@ import "../../libraries/DisputeTypes.sol";
...
-     *         `gameType ++ rootClaim ++ extraData` to the deployed `DisputeGame` clone.
-     * @dev `++` equates to concatenation.
-     * @param gameType The type of the DisputeGame - used to decide the proxy implementation
-     * @param rootClaim The root claim of the DisputeGame.
-     * @param extraData Any extra data that should be provided to the created dispute game.
-     * @return _proxy The clone of the `DisputeGame` created with the given parameters.
-     *         Returns `address(0)` if nonexistent.
-     * @return _timestamp The timestamp of the creation of the dispute game.
-     */
+    /// @notice `games` queries an internal a mapping that maps the hash of
```
Comment (refcell) [inline packages/contracts-bedrock/contracts/dispute/interfaces/IDisputeGameFactory.sol:30]: Small nit, surprised we missed this b4
```suggestion
    /// @notice `games` queries an internal mapping that maps the hash of
```

File: packages/contracts-bedrock/contracts/dispute/interfaces/IDisputeGameFactory.sol [hunk 3]
```diff
@@ -5,105 +5,87 @@ import "../../libraries/DisputeTypes.sol";
...
 
-    /**
-     * @notice Creates a new DisputeGame proxy contract.
-     * @param gameType The type of the DisputeGame - used to decide the proxy implementation
-     * @param rootClaim The root claim of the DisputeGame.
-     * @param extraData Any extra data that should be provided to the created dispute game.
-     * @return proxy The address of the created DisputeGame proxy.
-     */
+    /// @notice Creates a new DisputeGame proxy contract.
+    /// @param gameType The type of the DisputeGame - used to decide the proxy implementation
```
Comment (refcell) [inline packages/contracts-bedrock/contracts/dispute/interfaces/IDisputeGameFactory.sol:64]: I don't think we've strictly followed, but would be nice to fully punctuate comments
```suggestion
    /// @param gameType The type of the DisputeGame - used to decide the proxy implementation.
```

File: packages/contracts-bedrock/contracts/dispute/interfaces/IDisputeGameFactory.sol [hunk 4]
```diff
@@ -5,105 +5,87 @@ import "../../libraries/DisputeTypes.sol";
...
-     * @dev May only be called by the `owner`.
-     * @param gameType The type of the DisputeGame.
-     * @param impl The implementation contract for the given `GameType`.
-     */
+    ///
+    /// @notice Sets the implementation contract for a specific `GameType`.
+    /// @dev May only be called by the `owner`.
+    /// @param gameType The type of the DisputeGame.
+    /// @param impl The implementation contract for the given `GameType`.
+    ///
```
Comment (refcell) [inline packages/contracts-bedrock/contracts/dispute/interfaces/IDisputeGameFactory.sol:79]:
```suggestion
    /// @notice Sets the implementation contract for a specific `GameType`.
    /// @dev May only be called by the `owner`.
    /// @param gameType The type of the DisputeGame.
    /// @param impl The implementation contract for the given `GameType`.
```

File: packages/contracts-bedrock/contracts/dispute/interfaces/IDisputeGameFactory.sol [hunk 5]
```diff
@@ -5,105 +5,87 @@ import "../../libraries/DisputeTypes.sol";
...
-     * @param extraData Any extra data that should be provided to the created dispute game.
-     * @return _uuid The unique identifier for the given dispute game parameters.
-     */
+    /// @notice Returns a unique identifier for the given dispute game parameters.
+    /// @dev Hashes the concatenation of `gameType . rootClaim . extraData`
+    ///      without expanding memory.
+    /// @param gameType The type of the DisputeGame.
+    /// @param rootClaim The root claim of the DisputeGame.
+    /// @param extraData Any extra data that should be provided to the created dispute game.
+    /// @return _uuid The unique identifier for the given dispute game parameters.
```
Comment (refcell) [inline packages/contracts-bedrock/contracts/dispute/interfaces/IDisputeGameFactory.sol:88]: Maybe rip into another pr so we don't have to regen gas snapshot, bindings, etc etc, but would be nice to use the underscore postfix return standard

File: packages/contracts-bedrock/contracts/test/LibPosition.t.sol [hunk 6]
```diff
@@ -88,10 +74,8 @@ contract LibPosition_Test is Test {
         assertEq(parent.indexAtDepth(), _indexAtDepth / 2);
     }
 
-    /**
-     * @notice Tests that the `rightIndex` function correctly computes the deepest, right most index relative
-     * to a given position.
-     */
+    /// @notice Tests that the `rightIndex` function correctly computes the deepest, right most index relative
+    /// to a given position.
```
Comment (refcell) [inline packages/contracts-bedrock/contracts/test/LibPosition.t.sol:78]: Small nit (think this is aligned right - hard to check on gh :/)
```suggestion
    /// @notice Tests that the `rightIndex` function correctly computes the deepest, right most index relative
    ///         to a given position.
```

File: packages/contracts-bedrock/contracts/test/LibPosition.t.sol [hunk 7]
```diff
@@ -115,11 +99,9 @@ contract LibPosition_Test is Test {
         assertEq(Position.unwrap(rightIndex), Position.unwrap(position));
     }
 
-    /**
-     * @notice Tests that the `attack` function correctly computes the position of the attack relative to
-     * a given position.
-     * @dev `attack` is an alias for `left`, but we test it separately for completeness.
-     */
+    /// @notice Tests that the `attack` function correctly computes the position of the attack relative to
+    /// a given position.
```
Comment (refcell) [inline packages/contracts-bedrock/contracts/test/LibPosition.t.sol:103]: ibid

File: packages/contracts-bedrock/contracts/test/LibPosition.t.sol [hunk 8]
```diff
@@ -132,12 +114,10 @@ contract LibPosition_Test is Test {
...
-    /**
-     * @notice Tests that the `defend` function correctly computes the position of the defense relative to
-     * a given position.
-     * @dev A defense can only be given if the position does not belong to the root claim, hence the bound of [1, 127]
-     * on the depth.
-     */
+    /// @notice Tests that the `defend` function correctly computes the position of the defense relative to
+    /// a given position.
+    /// @dev A defense can only be given if the position does not belong to the root claim, hence the bound of [1, 127]
+    /// on the depth.
```
Comment (refcell) [inline packages/contracts-bedrock/contracts/test/LibPosition.t.sol:120]: ibid

**Review by refcell — CHANGES_REQUESTED** [review]
Just small styling changes

File: packages/contracts-bedrock/contracts/dispute/interfaces/IDisputeGameFactory.sol (hunk 5 above)
Comment (clabby) [reply packages/contracts-bedrock/contracts/dispute/interfaces/IDisputeGameFactory.sol:86]: For sure, can do that in a separate PR.

**Review by refcell — APPROVED** [review]

**Comment by clabby** [comment]
@mergifyio queue

**Pull request merged**